### PR TITLE
devstack: Add option to enable fake virt driver for nova

### PR DESCRIPTION
**Pull request by gndrmnn**
Used in https://github.com/osism/openstack-simple-stress/pull/63

Tested in https://zuul.services.betacloud.xyz/t/osism/build/ebd5dfc6a4ea469088d98951afd6d1d9

### CODE REVIEW

File: roles/devstack/tasks/main.yml [hunk 1]
```diff
@@ -64,6 +70,8 @@
       RABBIT_PASSWORD={{ devstack_password }}
       SERVICE_PASSWORD={{ devstack_password }}
 
+      {{ devstack_use_fake_driver_command }}
```
Comment (berendt) [inline roles/devstack/tasks/main.yml:73]: Use `devstack_use_fake_driver_command | default('')` here.

File: roles/devstack/defaults/main.yml [hunk 2]
```diff
@@ -10,3 +10,7 @@ devstack_password: password
 
 # Services to deploy
 devstack_services: "keystone g-api"
+
+# Use this option to enable the nova fake virt driver for testing
+devstack_use_fake_driver: false
+devstack_use_fake_driver_command: ""
```
Comment (berendt) [inline roles/devstack/defaults/main.yml:16]: Remove this.

**Review by berendt — CHANGES_REQUESTED** [review]

**Comment by gndrmnn** [comment]
@berendt done :)

File: roles/devstack/tasks/main.yml [hunk 3]
```diff
@@ -50,6 +50,12 @@
     executable: /bin/bash
   changed_when: false
 
+- name: Setup fake virt driver command
+  ansible.builtin.set_fact:
+    devstack_use_fake_driver_command: "VIRT_DRIVER=fake"
+  when: devstack_use_fake_driver
+  changed_when: false
```
[the remark this answers is not in the document]
Comment (gndrmnn) [reply roles/devstack/tasks/main.yml:57]: Do you mean the ``changed_when``?

[the remark this answers is not in the document]
Comment (gndrmnn) [reply roles/devstack/tasks/main.yml:57]: Or do you want a ``{{ "VIRT_DRIVER=fake" if devstack_use_fake_driver else ""  }}`` and omit the ``devstack_use_fake_driver_command`` entirely?

[the remark this answers is not in the document]
Comment (berendt) [reply roles/devstack/tasks/main.yml:57]: Jop, that's much prettier.

**Pull request merged**